src/battleship/BattleshipGameRunner.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace battleship
+{
+    public interface IBattleshipGameRunner
+    {
+        void RunGame();
+    }
+
+    public class BattleshipGameRunner : IBattleshipGameRunner
+    {
+        private readonly IGameBoardInitializer gameBoardInitializer;
+        private readonly ICoordinateInput coordinateInput;
+
+        public BattleshipGameRunner(IGameBoardInitializer gameBoardInitializer, ICoordinateInput coordinateInput)
+        {
+            this.gameBoardInitializer = gameBoardInitializer;
+            this.coordinateInput = coordinateInput;
+        }
+
+        public void RunGame()
+        {
+            var gameBoard = gameBoardInitializer.PrepareGame();
+
+            while (gameBoard.Winner is null)
+            {
+                try
+                {
+                    gameBoard = gameBoard.PlayTurn((p) =>
+                    {
+                        Console.WriteLine($"It is {p.Name}'s turn");
+                        // Todo: print additionnal contextual data like previous shots taken.
+                        return coordinateInput.GetCoordinate();
+                    });
+                }
+                catch (ArgumentOutOfRangeException ex)
+                {
+                    Console.WriteLine(ex.Message);
```

Comment:
Here the error message will not be user friendly in the console. I could use custom exceptions to have a better control over the message